avogadro/core/array.h

```diff
@@ -118,19 +121,18 @@ class Array
 
   explicit Array(const size_t n, const ValueType& value = ValueType())
     : d(new Container(n, value))
-  {}
+  {
+  }
 
   template <typename InputIterator>
   Array(InputIterator first, InputIterator last) : d(new Container(first, last))
-  {}
-
-  /** Copy constructor, note the copy made of the internal data of other. */
-  Array(const Array& other)
   {
-    other.d->reref();
-    d = other.d;
   }
 
+  /** Copy constructor, note the copy made of the internal data of other. */
+  Array(const Array& other) : d(new Container(*other.d)) {}
```

Comment:
Yeah, we don't want to do this. The Array class is copy-on-write. Please remove and/or tell cppcheck to ignore this.

Reply:
ok